packages/angular/cli/src/utilities/completion.ts

```diff
@@ -188,6 +193,11 @@ will modify configuration files in your home directory.)
  * @return The full path of the configuration file modified.
  */
 export async function initializeAutocomplete(): Promise<string> {
+  // Windows doesn't support autocompletion, give a clear error message.
+  if (process.platform === 'win32') {
+    throw new Error(`Windows does not support shell autocompletion.`);
```

Comment:
While I haven't tried it, I don't think this is correct. Auto complete should work on Windows as long as the user has bash installed, example through git bash.

Reply:
I'm not sure about git bash but WSL 1/2 should have a `platform` of `linux` so that will be covered with that check.

Reply:
I am referring to usages outside of WSL. 

On native Windows, zsh and bash can still be installed and available and autocompletion is still possible using;

Example using;
- cygwin
- cmder
- git bash

Considering a large set of our users are on Windows OS, I think we should support this case better. Alternatively, for the time being we can say that the automated setup doesn't support Windows but explain how this can be done by modifying the zsh and bash profiles manually.

Reply:
We could also check the `OSTYPE` environment variable when on Windows.  I believe it should be `msys` or `cygwin` for those cases but would need to confirm.

Reply:
Yeah, I was going to check Git Bash today, as I suspect that will be `win32`. I forgot about Cygwin, but I could test that too. Cmder I thought was just a terminal which still used `cmd` under the hood and would be `win32` (it's not a Linux emulator IIRC)? WSL should definitely be detected as Linux.

I'm wondering if a better approach might be to check `$SHELL` for `cmd.exe` or `powershell.exe` and print an error message on that case? Not sure if there are other Windows shells which might come up there.

Reply:
Just checked my personal machine and `process.platform` + `$SHELL` for each environment is:

* Cmd has platform `win32` and `$SHELL` is unset - should fail with a useful error message.
* Powershell has platform `win32` and `$SHELL` is unset - should fail with a useful error message.
* WSL2 has platform `linux` and shell `/usr/bin/zsh` - "just works".
* Git Bash has platform `win32` and shell `C:\Program files\Git\usr\bin\bash.exe` - NVM autocompletion works, so I suspect we should too, but I wasn't able to get it to use a local CLI build to confirm.
* Cmder has platform `win32` and `$SHELL` is unset, but I do see `$CMDER_SHELL="cmd"` - I suspect this shouldn't work and we should fail with a useful error message.
* Cygwin couldn't install Node in a reasonable time frame, so I gave up there -  I suspect this is similar to Git Bash and probably falls into that case.

All the unsupported systems have an unset `$SHELL`, so I think that is the best way of detecting the failure. Unfortunately there is no good way to disambiguate between the environments (Powershell doesn't seem to have any good environment variable to identify itself for instance). I'm inclined to fail with a more generic message on missing `$SHELL` along the lines of:

> `$SHELL` environment variable not set. Only Bash and Zsh are supported. If you're on Windows, Cmd and Powershell don't support command autocompletion, but Git Bash or Windows Subsystem for Linux should work, so make sure to run the command from there.
> 
> https://angular.io/cli/some-doc-about-autocompletion

(Sidebar: We should have a more directed/"goal-oriented" doc for setting up autocompletion which can explain the supported shells and link to relevant pieces.)